controllers/shooter.py

```diff
@@ -1,91 +1,85 @@
 
 
 class ShooterController(StateMachine):
+    chassis_component: Chassis
     intake_component: Intake
     shooter_component: Shooter
     tilt_component: Tilt
     turret_component: Turret
 
     def __init__(self) -> None:
-        # set default preference for goal height
-        pass
+        self.goal_height_preference = GoalHeight.HIGH  # default preference
+
+    try_shoot = will_reset_to(False)
 
     @state(first=True, must_finish=True)
     def preparing_intake(self) -> None:
-        # send pitch and turret to angle
+        self.turret_component.set_angle(0.0)
+        self.tilt_component.goto_intaking()
 
-        # if at pitch and turret angle
-        # next state intaking
-        pass
+        if self.turret_component.at_angle() and self.tilt_component.at_angle():
+            self.next_state("intaking")
 
     @state(must_finish=True)
     def intaking(self) -> None:
-        # shooter intake
-        # intake deploy
+        self.intake_component.deploy()
 
-        # if shooter is loaded
-        # retract intake
-        # next state tracking
-        pass
+        if self.shooter_component.is_loaded():
+            self.intake_component.retract()
+            self.next_state("tracking")
 
     @default_state
     def tracking(self) -> None:
-        # if turret not indexed
-        # return
-
-        # update flywheel speed
-        # update turret angle
-        # update tilt angle
+        self.update_component_setpoints()
 
-        pass
+        if (
+            self.try_shoot
+            and self.shooter_component.is_loaded()
+            and self.turret_component.at_angle()
+            and self.tilt_component.at_angle()
+        ):  # and self.shooter_component.at_speed()
+            self.next_state("shooting")
 
-    @state(must_finish=True)
+    @timed_state(must_finish=True, duration=1.0, next_state="tracking")
     def shooting(self) -> None:
-        # update flywheel speed
-        # update turret angle
-        # update tilt angle
-
-        # if tilt ready and turret ready and flywheels at speed
-        # shoot
-        # next state tracking
-        pass
+        self.update_component_setpoints()
+        self.shooter_component.shoot()
+
+    def update_component_setpoints(self) -> None:
+        bs = calculate_ballistics(
+            self.chassis_component.get_pose(), Pose2d(), self.goal_height_preference
+        )
+        self.shooter_component.set_flywheel_speed(
+            bs.top_flywheel_speed, bs.bottom_flywheel_speed
+        )
+        self.turret_component.set_angle(bs.turret_angle)
+        self.tilt_component.set_angle(bs.tilt_angle)
 
     def shoot(self) -> None:
-        # set next state to shooting
-        pass
+        self.try_shoot = True
 
     def intake(self) -> None:
-        # if shooter doesn't have a cube
-        # set next state to preparing intake
-        pass
+        if not self.shooter_component.is_loaded():
+            self.next_state("preparing_intake")
```

Comment:
This should just engage the state machine, rather than forcibly preparing to intake, since we shouldn't attempt to intake whilst shooting.
```suggestion
            self.engage()
```